src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -1318,6 +1318,23 @@ public void testAdjacentIndels() {
                 .allMatch(vc -> vc.isBiallelic() && (vc.getReference().length() == 1 || vc.getAlternateAllele(0).length() == 1)));
     }
 
+    // this test has a reference with 8 repeats of a 28-mer and an alt with 7 repeats.  This deletion left-aligns to the
+    // beginning of the padded assembly region, and an exception occurs if we carelessly drop the leading deletion from
+    // the alt haplotype's cigar.  This is a regression test for https://github.com/broadinstitute/gatk/issues/6533.
+    @Test
+    public void testLeadingDeletionInAltHaplotype() {
+        final File bam = new File(TEST_FILES_DIR, "alt-haplotype-leading-deletion.bam");
+
```

Comment:
I like that we have an integration test. Generally even if the test is intended to assert that we aren't getting an exception its best to actually assert something about the output. It wouldn't be too difficult to check in an output VCF for this file I shouldn't think.

Reply:
Added check that the 28-base deletion is called.